controllers/util/condition.go

```diff
@@ -75,6 +74,13 @@ func SetReconcileErrorCondition(conditions *[]odfv1alpha1.Condition, reason stri
 		Reason:  reason,
 		Message: message,
 	})
+
+	SetStatusCondition(conditions, odfv1alpha1.Condition{
```

Comment:
I saw we call to `SetReconcileErrorCondition` in many cases other than missing CSI CR, so I am not sure we need to change this status here.

Reply:
added SetReconcileProvisionerErrorCondition